src/gti.jl

```diff
@@ -1,3 +1,194 @@
+function get_total_gti_length(gti::AbstractMatrix{<:Real}; minlen::Real=0.0)
+    lengths = diff(gti; dims =2)
+    return sum(x->x > minlen ? x : zero(x), lengths)
+end
+
+function load_gtis(fits_file::String, gtistring::String="GTI")
+    lchdulist = FITS(fits_file)
+    gtihdu = lchdulist[gtistring]
+    gti = get_gti_from_hdu(gtihdu)
+    close(lchdulist)
+    return gti
+end
+
+function get_gti_from_hdu(gtihdu::TableHDU)
+
+    if "START" in FITSIO.colnames(gtihdu)
+        startstr = "START"
+        stopstr = "STOP"
+    else
+        startstr = "Start"
+        stopstr = "Stop"
+    end
+
+    gtistart = read(gtihdu,startstr)
+    gtistop = read(gtihdu,stopstr)
+
+    return mapreduce(permutedims, vcat, 
+    [[a, b] for (a,b) in zip(gtistart, gtistop)])
+end
+
+function check_gtis(gti::AbstractMatrix)
+
+    if ndims(gti) != 2 || size(gti,2) != 2
+        throw(ArgumentError("Please check the formatting of the GTIs. 
+        They need to be provided as [[gti00 gti01]; [gti10 gti11]; ...]."))
+    end
+
+    gti_start = @view gti[:, 1]
+    gti_end = @view gti[:, 2]
+
+    if any(gti_end < gti_start)
+        throw(ArgumentError(
+            "The GTI end times must be larger than the GTI start times."
+        )) 
+    end
+
+    if any(@view(gti_start[2:end]) < @view(gti_end[1:end-1]))
+        throw(ArgumentError(
+            "This GTI has overlaps"
+        ))
+    end
+end
+
+function create_gti_mask(times::AbstractVector{<:Real},gtis::AbstractMatrix{<:Real};
+                         safe_interval::AbstractVector{<:Real}=[0,0], min_length::Real=0,
+                         dt::Real = -1, epsilon::Real = 0.001)
+
+    if length(times) == 0
```

Comment:
```suggestion
    if isempty(times)
```
should be the same, right?